File: 2020/python/Day01_Part2.py

def main():
    expense_report = ReadPuzzleInput("PuzzleInput_Day01.txt")
    ints = BruteForce(expense_report)
    print("the answer is:", ints[0], "*", ints[1], "*", ints[2], "=", ints[0]*ints[1]*ints[2])

def ReadPuzzleInput(my_file):
    with open(my_file, 'r') as f:
        return [int(line) for line in f]

def BruteForce(array):
    for a in array:
        for b in array:
            for c in array:
                if (a + b + c == 2020):
                    return [a,b,c]

if __name__ == "__main__":
    main()

File: 2020/Java/Day01_Part2.java
import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;
import java.io.File;

public class Day01_Part2 {
        public static void main(String[] args) {
            // read input
            List<Integer> expense_report = new ArrayList<>();
            try {
                Scanner scanner = new Scanner(new File("PuzzleInput_Day01.txt"));
                while(scanner.hasNextInt())
                {
                    expense_report.add(scanner.nextInt());
                }
            } catch(Exception e) {
                System.out.println(e);
            }
            // bruteforce
            for (Integer a : expense_report) {
                for (Integer b : expense_report) {
                    for (Integer c : expense_report) {
                        if (a + b + c == 2020) {
                            System.out.println(a*b*c);
                            return;
                        }
                    }
                }
            }
        }
}
